surfactant/cmd/cli_commands/cli_base.py

```diff
@@ -0,0 +1,88 @@
+import dataclasses
+import os
+import pickle
+import platform
+from dataclasses import Field
+from pathlib import Path
+
+from loguru import logger
+
+from surfactant.sbomtypes._sbom import SBOM
+
+
+class Cli:
+    """
+    A base class that implements the surfactant cli basic functionality
+
+    Attributes:
+    sbom                    An internal record of sbom entries the class adds to as it finds more matches.
+    subset                  An internal record of the subset of sbom entries from the last cli find call.
+    sbom_filename:          A string value of the filename where the loaded sbom is stored.
+    subset_filename:        A string value of the filename where the current subset result from the "cli find" command is stored.
+    match_functions         A dictionary of functions that provide matching functionality for given SBOM fields (i.e. uuid, sha256, installpath, etc)
+    camel_case_conversions  A dictionary of string conversions from all lowercase to camelcase. Used to convert python click options to match the SBOM attribute's case
+
+    """
+
+    sbom: SBOM = None
+    subset: SBOM = None
+    sbom_filename: str
+    subset_filename: str
+    match_functions: dict
+    camel_case_conversions: dict
+
+    def __init__(self):
+        self.sbom_filename = "sbom_cli"
+        self.subset_filename = "subset_cli"
+        # Create data directory
+        self.data_dir = self._get_cli_sbom_dir()
```

Comment:
Can the function in the ConfigManager (https://github.com/LLNL/Surfactant/blob/main/surfactant/configmanager.py#L145) get used instead? I think the one difference is it uses Local AppData on Windows -- it doesn't get synchronized between multiple systems, which is probably okay (esp. if the SBOM happens to be very big).

Reply:
Got it updated! If it needs to be changed at any point, we can integrate a new location into the ConfigManger then